example/lib/device_details/device_details_bloc.dart

```diff
@@ -112,6 +114,39 @@ class DeviceDetailsBloc {
         })
         .then((characteristics) => characteristics.forEach((characteristic) =>
             log("Found characteristic \n ${characteristic.uuid}")))
+        .then((_) {
+          log("Turn off temperature update");
+          return peripheral.writeCharacteristic(
```

Comment:
You're demoing only one approach, ie. reading/writing from peripheral level using known UUIDs. You should also test against those operation from Service and Characteristic level.